raiden-ts/tests/unit/epics/channels.spec.ts

```diff
@@ -919,3 +920,21 @@ describe('channelUnlockEpic', () => {
     );
   });
 });
+
+test('channelAutoSettleEpic', async () => {
+  expect.assertions(2);
+
+  const [raiden, partner] = await makeRaidens(2);
+
+  // enable autoSettle
+  raiden.store.dispatch(raidenConfigUpdate({ autoSettle: true }));
+
+  await ensureChannelIsClosed([raiden, partner]);
+  await waitBlock(settleBlock + 1);
+  await waitBlock(settleBlock + 2 * confirmationBlocks + 1);
+
+  expect(raiden.output).toContainEqual(
+    channelSettle.request(undefined, { tokenNetwork, partner: partner.address }),
+  );
+  expect(getChannel(raiden, partner)?.state).toBe(ChannelState.settling);
```

Comment:
I don't think this works. I think there could be a race with the processing of blocks. I would not guarantee that this test is not flaky due to the randomness and the off plus one range.
Rather I would wait for the settleBlock and afterwards to test if we see the according actions. I think this would be more reliable. But I must admit that I'm not 100% sure about this here. Could fail on both ends. 
Or wait a second: does Math.random() produces number in `[0,1]` or `]0,1[`?

Reply:
This works because `channelSettle.request` is reduced syncronously, and it already sets the channel.state to `settling`, no need to wait for the tx.
`Math.random()` produces numbers in the range `[0, 1[` AFAIK, but it doesn't matter, we `Math.round` after multiplication.

Reply:
Hmm okay that is true. But why does the channel not switch to state `settled`? :thinking:

Reply:
Because it only goes to that state after `settleChannel` tx is confirmed, not right after the request. Also, sync.

Reply:
Let's close this.  If it ever fails for random `0` we know why. Augusto just sat here yesterday and also wondered about the test.